src/client/address_book/storage.cc

```diff
@@ -65,18 +67,34 @@ bool AddressBookStorage::GetAddress(
   return true;
 }
 
-void AddressBookStorage::AddAddress(
-    const kovri::core::IdentityEx& address) {
-  auto filename = GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
-  std::ofstream file(filename.string(), std::ofstream::binary);
-  if (!file)
-    LOG(error) << "AddressBookStorage: can't open file " << filename;
+void AddressBookStorage::AddAddress(const kovri::core::IdentityEx& address)
+{
+  auto filename =
+      GetAddressesPath() / (address.GetIdentHash().ToBase32() + ".b32");
+  kovri::core::OutputFileStream file(filename.string(), std::ofstream::binary);
+  if (!file.Good())
+    {
+      m_Exception.Dispatch(__func__);
+      throw std::runtime_error("failed to open file for writing");
+    }
+  // For sanity, triple-check the validity of the identity length
```

Comment:
This is entirely unnecessary, this check should be removed. Identity checks must be done within identity code.